source/_posts/angular-signals.md

```diff
@@ -0,0 +1,409 @@
+
+What about `ChangeDetectionStrategy.OnPush`, which serves as a "smart change detection checker" that improves our application performance? Won't it be redundant when using signals?
+
+{% add_classes "OnPush change detection strategy" "center bold" %}
+{% img "change-detection-strategy-onpush.png" "OnPush change detection" "lazy" %}
+
+It seems that `OnPush` can be unnecessary. Why? Let's imagine that we don't have Zone.js and everywhere in the application we only have scattered signals and their dependencies that update themselves in a granular way. At this point, we don't need to disable any components from dirty checking in the tree anymore as we did in the `OnPush` strategy. We use signals inside and outside the components. And that's it. Everything updates automatically.
+
+## Will Angular Signals replace RxJS?
+
+As we know Angular married RxJS from the very beginning. Rxjs is the core library that is responsible for reactivity in Angular. It is a great library with which you can overcome many challenges easily and quickly. On top of that, it is very pleasant to write Angular code using:
+
+* streams,
+* [cold & hot observables](https://courses.dev-academy.com/p/advanced-rxjs),
+* subjects,
+* async pipes,
+* operators,
+* [caching](http://localhost:4000/angular-cache-http-requests/)
+* and many more.
+
+We know Angular mainly uses RxJS, so we can easily deal with asynchronicity and data passing between components. By the way, if you need to learn about the advanced techniques of RxJS then I encourage you to take a look here at this course: *[Master advanced RxJS ☄️ techniques](https://courses.dev-academy.com/p/advanced-rxjs)*.
+
+{% add_classes "Will signals replace Rxjs in some way?" "bold" %}
+
+It seems so, but only on certain things. For example, let's consider the Subjects in RxJS. If we compare Angular Signals with Subjects, we can observe some key differences.
+1. **Much simpler than streams**
+   Simpler and more understandable code. It means it is easier to get into Angular and learn it. No need to learn RxJS at the start, just start using signals.
+
+2. **No need to handle subscription/unsubscription**
+   Just like the `async` pipe in the template. You don't have to manually `unsubscribe`. This is what the Angular team said:
+   
+   > Implicit, low-overhead subscriptions
+
+3. **Automatic propagation of changes in each dependency**
+   No need to trigger any updates in the code. That *reactive primitive* has a dependency tree that automatically synchronizes itself.
+
+Overall, I don't think there is much to fear about Signals being able to displace RxJS. There is no basis for this. The Angular team has stated that they are going to *integrate Signals into RxJS*.
+
+I think RxJS with Signals will be a great combination from which we will be able to benefit even more!
+
+## Benefits of using Angular Signals
+
+- a new reactive primitive,
+- simplify the framework,
+- **granular updates**,
```

Comment:
If you want this as something of more importance, maybe put it first instead of bolding it. Does not look good to me on the rendered page.

Reply:
Good point!